parsec/core/gui/files_widget.py

```diff
@@ -107,9 +108,15 @@ async def _do_import(workspace_fs, files, total_size, progress_signal):
                         progress_signal.emit(src.name, current_size + read_size)
             current_size += read_size + 1
             progress_signal.emit(src.name, current_size)
-
         except trio.Cancelled as exc:
-            raise JobResultError("cancelled", last_file=dst) from exc
+            errors.append(exc)
+            raise JobResultError(
+                "cancelled", last_file=dst, file_count=len(files), exceptions=errors
+            ) from exc
+        except PermissionError as exc:
+            errors.append(exc)
```

Comment:
It should be possible to remove the if statement below and raise the exception in the `except PermissionError as exc` statement no ?

Reply:
This one has bothered me for some time. Since we can be importing more than one file, we have two options:
- we stop at the first error and not even try to import some files
- we try to import as many files as possible

I chose the later even though I'm not convinced that it's the best. I'll open an issue on the subject so we can discuss it.